lib/TPP/TileConsumerAndFuseProducers.cpp

```diff
@@ -140,10 +272,30 @@ struct TileConsumerAndFuseProducers
     linalg::registerTilingInterfaceExternalModels(registry);
   }
   void runOnOperation() override {
+    func::FuncOp func = getOperation();
+    transform::TrivialPatternRewriter rewriter(&getContext());
+    llvm::SmallDenseSet<Operation *> fusedOps;
+    func->walk([&](linalg::LinalgOp linalgOp) {
+      if (linalg::isElementwise(linalgOp) && hasMatmulLikeProducer(linalgOp)) {
```

Comment:
Nothing special but it is clearly capable of fusing chain of matmuls or other ops. So, it should be able to kick in without an element-wise op. Or will that break some current assumptions?

Reply:
It seems `hasMatmulLikeProducer` only checks for immediate producers, so a `matmul -> ew1 -> ew2` won't fuse `ew2` since it doesn't have `matmul` producers (and this check will fail).

The test `@matmul_sequence_fusion` below shows that the final `generic (addf)` doesn't get fused with the bundle, because its producer is a `generic (maxf)`. But if the tiling spec is the same, then there's no reason for that not to be fused, even if the producer isn't a matmul.

Reply:
It probably could iterate greedily over the whole chain of ops and only stop when you cannot fuse the next producer/consumer. As there could be some pitfalls to this approach, I guess it could be investigated separately and we can keep the simpler anchor approach for this PR.

Reply:
I think this is complex enough that deserves to be in a separate PR, and with more discussion around it.